code_generation/producer.py

```diff
@@ -271,3 +273,13 @@ def writecalls(self, config, scope):
             # retrieve calls of subproducers
             calls.extend(producer.writecalls(config, scope))
         return calls
+
+
+class Filter(ProducerGroup):
```

Comment:
Should this be a Group Filter, rather than a filter only ?

Reply:
yes, I thoght that would be useful since every filter will contain the actual DF-Filterstep and a producer of the input flags. This saves us an instance of ProducerGroup that just collects the two parts.

Reply:
Should we also change this to `FilterFlags` than ? Or stay with Filter ? I think both is fine

Reply:
I think this is the only usecase for the python-filter. That's why I turned the previous Filter into BaseFilter and called this one here Filter.
For simpler Filters without shift like the MET-Filter the simple Producer was sufficient.
But actually we could also use the BaseFilter (or the Filter without subproducers would also work) just to avoid confusion by the naming.